pkg/scheduler/objects/application.go

```diff
@@ -1508,6 +1510,9 @@ func (sa *Application) addAllocationInternal(info *Allocation) {
 		sa.allocatedResource = resources.Add(sa.allocatedResource, info.GetAllocatedResource())
 		sa.maxAllocatedResource = resources.ComponentWiseMax(sa.allocatedResource, sa.maxAllocatedResource)
 	}
+	if info.GetPriority() < sa.allocMinPriority {
```

Comment:
add a variable here so we don't call GetPriority() multiple times (this locks the object repeatedly)